clients/p4rt-ctl/p4rt-ctl.in

```diff
@@ -769,6 +791,52 @@ def p4ctl_show(client, bridge):
 def p4ctl_set_pipe(client, bridge, device_config, p4info):
     client.set_fwd_pipe_config(p4info, device_config)
 
+class TapDevice:
+    def __init__(self, tap_name):
+        TUNSETIFF = 0x400454CA
```

Comment:
Where do these values come from? Is there a spec that users can look at (is that open sourced or internal only)?

Also, this feature is only supported on ES2K target. Perhaps we should add a comment somewhere so that it is clear that only ES2K target is supported?

Reply:
These macros are defined in 
https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/if_tun.h#L34
But I got them from sample programs available online.

PacketIO is supported by Tofino as well.
The messages(PacketIn and PacketOut) used are supported by p4runtime so the client and server can exchange these streaming messages irrespective of the target.
I'm not sure if we need to add the limitation in the client code as it is not really this code's limitation.